cli/commands/queueWithdrawal.go

```diff
@@ -41,15 +43,38 @@ func QueueWithdrawalCommand(args TQueueWithdrawallArgs) error {
 
 	_reg, err := pod.WithdrawableRestakedExecutionLayerGwei(nil)
 	core.PanicOnError("failed to load REG", err)
-	reg := new(big.Int).SetUint64(_reg)
+
+	// [withdrawable]RestakedExecutionlayerWei
+	rew := core.GweiToWei(new(big.Float).SetUint64(_reg))
+	if args.AmountWei > 0 && new(big.Float).SetUint64(args.AmountWei).Cmp(rew) > 0 {
+		return errors.New("invalid --amountWei. must be in the range (0, pod.withdrawableRestakedExecutionLayerGwei() as wei]")
+	}
 
 	podOwner, err := pod.PodOwner(nil)
 	core.PanicOnError("failed to read podOwner", err)
 
 	res, err := dm.GetWithdrawableShares(nil, podOwner, []common.Address{core.BeaconStrategy()})
 	core.PanicOnError("failed to read beacon strategy withdrawable shares", err)
 
-	_depositShares, err := dm.ConvertToDepositShares(nil, podOwner, []common.Address{core.BeaconStrategy()}, []*big.Int{res.WithdrawableShares[0]})
+	reg := new(big.Int).SetUint64(_reg)
+
+	requestedWithdrawalSizeWei := func() *big.Int {
+		if args.AmountWei == 0 {
+			// if AmountWei isn't specified, we withdraw all the shares in the beacon strategy.
+			return res.WithdrawableShares[0]
+		}
+
+		// if it is specified, we withdraw the specific amount.
+		return new(big.Int).SetUint64(args.AmountWei)
+	}()
+	requestedWithdrawalSizeGwei := core.WeiToGwei(requestedWithdrawalSizeWei)
+
+	if requestedWithdrawalSizeGwei.Cmp(new(big.Float).SetInt(res.WithdrawableShares[0])) > 0 {
```

Comment:
we should limit these to

rew - sum(queued withdrawals)

Reply:
done